src/components/ColorPicker/components/ColorPickerContent/ColorPickerClearButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useCallback } from "react";
+import useGridKeyboardNavigation from "../../../../hooks/useGridKeyboardNavigation/useGridKeyboardNavigation";
+import Button from "../../../Button/Button";
+import { VibeComponentProps, VibeComponent } from "../../../../types";
+import { NOOP } from "../../../../utils/function-utils";
+
+interface ColorPickerClearButtonProps extends VibeComponentProps {
+  onClick: (e: React.MouseEvent<HTMLButtonElement>) => any;
+  text?: string;
+  Icon: ({ size, className }: { size?: string; className?: string }) => JSX.Element;
+}
+
+export const ColorPickerClearButton: VibeComponent<ColorPickerClearButtonProps> = React.forwardRef(
+  ({ onClick, text, Icon }, ref) => {
+    const onItemClicked = useCallback(() => onClick(null), [onClick]);
+
+    const { onSelectionAction } = useGridKeyboardNavigation({
+      ref: ref as React.MutableRefObject<HTMLElement>,
+      itemsCount: 1,
+      numberOfItemsInLine: 1,
+      onItemClicked, // hack - we don't really have a grid, it's just for keyboard navigation outside the clear button
+      getItemByIndex: NOOP //hack - we don't really have a grid, it's just for keyboard navigation outside the clear button
+    });
+
+    return (
+      <Button
+        ref={ref}
+        size={Button.sizes.SMALL}
+        kind={Button.kinds.TERTIARY}
+        onClick={() => onSelectionAction(-1)} //hack - we don't really have a grid, it's just for keyboard navigation outside the clear button
```

Comment:
The Button's signature of `onClick` doesn't match to `onSelectionAction`.
Given that, `onSelectionAction` expects the item's index in the Grid but since there's only a single`ClearButton`, it can be anything

Reply:
Ok, sound good to me.
If that's the case, we can avoid re-creating anonymous function every time this component is rendered. Let's use useCallback hook for recreating the on click function only when necessary (only once when on this component first render)

Reply:
not using an anonymous function anymore